internal/auth.go

```diff
@@ -128,9 +128,17 @@ func redirectBase(r *http.Request) string {
 	return fmt.Sprintf("%s://%s", r.Header.Get("X-Forwarded-Proto"), r.Host)
 }
 
-// Return url
+// returnUrl is the URL that we will return to after authentication. Uses the `X-Forwarded-Prefix` header if provided,
+// since traefik just requests the root (i.e. r.URL.Path is almost always /)
 func returnUrl(r *http.Request) string {
-	return fmt.Sprintf("%s%s", redirectBase(r), r.URL.Path)
+	fwdUri := r.URL.Path
```

Comment:
I'm currently running into a related issue. For us we have in r.URL the correct page that we initially try to access. however if that URL has a query parameter, e.g. /myhorus/editor?open=123 then that ?open=123 is ommited and not there after going through authentication. Have you tried to r.URL.RequestURI for surch cases?

Reply:
I noticed that as a missing component 😞 

Unfortunately I don't think that will work without `traefik` changes. The problem is that `traefik` doesn't send along that information, so the `traefik-forward-auth` can't put it in the redirect and therefore the redirect can't restore it.

The problem is illustrated in the two links above. `traefik` just uses the "configured" address directly and does not include information about the original request. 

My suspicious is that to fix the query parameter issue, `traefik` would either need to:
1. Append the Path / Query Params onto the `GET` request for the forward auth check
2. Add a header that forwards the query parameters so that `traefik-forward-auth` can grab and use those headers

The other option is that it may be possible to write a custom forward auth middleware using the existing as a baseline, now that `traefik` has a mechanism for doing so. Forking `traefik` and building (2) would be pretty straightforward, I would think - I would expect (1) may be trickier.

Reply:
I can not confirm this. For me traefik is sending the actually requested URL to the traefik-forward-auth. I further changed the line as follows:
-       return fmt.Sprintf("%s%s", redirectBase(r), r.URL.Path)
+       return fmt.Sprintf("%s%s", redirectBase(r), r.URL.RequestURI())
Then the query part is also preserved.
I think you're lacking something else in your setup if it doesn't work for you. This is with traefik 2.6.6 for me

Reply:
Good to know!! I'll do a bit more testing. If that is the case, it will be welcome news to me! 😄

Reply:
You're right! `URL.RequestURI()` preserves the query... but not the path 😑 I'll try to see if I can figure out a hybrid approach

```
URL.Path: /
RequestURI: /
RequestURI(): /?query=hi
X-Forwarded-Prefix: /prefix
```

Reply:
Actually... it looks like `RequestURI()` can maintain some of the Request. Perhaps we have another middleware monkeying around here:
```
URL.Path: /hello/friend/
RequestURI: /
RequestURI(): /hello/friend/?query=hello
X-Forwarded-Prefix: /prefix

```

Reply:
If it helps, this is what we do in the middleware:
![image](https://user-images.githubusercontent.com/2754419/205725477-5716c51d-de9f-43e4-89b3-f375e3509d0c.png)

maybe it works for us because we do not strip anything away but just path the prefix through